src/vs/editor/common/viewModel/viewModel.ts

```diff
@@ -137,10 +141,10 @@ export class LineBreakData {
 		let delta = 0;
 		if (this.injectionOffsets !== null) {
 			for (let i = 0; i < this.injectionOffsets.length; i++) {
-				if (inputOffset < this.injectionOffsets[i]) {
+				if (inputOffset <= this.injectionOffsets[i]) {
```

Comment:
This looks like a change unrelated to the main goal of the PR. Why was this change needed?

Was this discovered by the new test in `src/vs/editor/test/common/viewModel/lineBreakData.test.ts`?

Reply:
> Was this discovered by the new test in src/vs/editor/test/common/viewModel/lineBreakData.test.ts?

Kind of. It is part of the commit "Improves cursor positioning for injected text.".
It makes the situation a little bit better until the general solution landed.

I discovered this when adding more tests.

Reply:
This might be one of those things that depends on the injected text being a `before` or `after` injected text.